### DISCUSSION TITLE
`wgrep` and Meow

### DISCUSSION THREAD

**Discussion by zetashift** [meow-edit/meow#585] [Q&A]
Does anybody have some sortof quick tutorial on how to use multicursor/beacon mode for Meow and `wgrep` in a Emacs buffer?
I'm a bit lost on:
- how to move around the MOTION grep buffer
- how to use multicursor to edit the buffer and save using `wgrep`

**Comment by Linerre** [comment]
Hi, sorry for the late reply. If you need to edit the buffer, then insertion mode is a better choice for the buffer's initial Meow state.
You can customize the `meow-mode-state-list`[^1] in your config, for example:

```
(add-to-list 'meow-mode-state-list '(rg-mode . normal))
```
which instructs Meow to enter the `rg-mode` buffer (showing the results of ripgrep search) with `normal-mode`.

[^1]: https://github.com/meow-edit/meow/blob/master/CUSTOMIZATIONS.org#meow-mode-state-list

**Reply by Linerre** [reply]
The above suggestion is generally fine if you want to enter a buffer (say, the results buffer of a grep search) with Meow's normal-mode.  But it's a bit misleading if we take into account the defaults provided by `rg.el` [^1], which offers a better approach:
1. Meow by default now enters the search results buffer with Motion mode
2. You are supposed to navigate through the buffer using the keybindings set for this buffer [^2], although you can also move the point with Emacs's default keybindings for movement [^3]
3. You can edit the buffer by pressing `e`  and save the buffer via `C-c C-c` [^4].  In my case, when I press `e`, the buffer automatically enables Meow normal mode. I have not configured anything about this behavior, and I'm happy with this default behavior 

[^1]: https://github.com/dajva/rg.el
[^2]: https://rgel.readthedocs.io/en/latest/usage.html#results-buffer
[^3]: https://www.gnu.org/software/emacs/manual/html_node/emacs/Moving-Point.html
[^4]: https://rgel.readthedocs.io/en/latest/usage.html#edit-and-apply-wgrep

Just in case whoever sees this in the future want to edit the search results buffer (of `rg`) with Meow

**Comment by DogLooksGood** [comment]
When you start wgrep, meow should switch to normal state automatically. Then you can do simple query/replace. Usually, you don't need beacon state here, if you do, just do it the same way as you do in normal text buffer.
